lib/core/tools/text_tool/text_tool.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:paintroid/core/commands/graphic_factory/graphic_factory.dart';
+import 'package:paintroid/core/tools/text_tool/bounding_box.dart';
+import 'package:paintroid/core/tools/tool.dart';
+
+class TextTool extends Tool {
+  bool isRotating;
+  BoundingBox boundingBox;
+
+  TextTool({
+    required super.commandManager,
+    required super.commandFactory,
+    required this.graphicFactory,
+    required super.type,
+    required this.boundingBox,
+    super.hasAddFunctionality = false,
+    super.hasFinalizeFunctionality = true,
+    this.isRotating = false,
+  });
+
+  String? currentText;
+  Offset? currentPosition;
+  bool isEditing = false;
+  final GraphicFactory graphicFactory;
+  Paint paint = Paint();
+
+  @override
+  void onDown(Offset point, Paint paint) {
+    paint = paint;
+    boundingBox.setActiveCorner(point, isRotating: isRotating);
+    isEditing = true;
+  }
+
+  @override
+  void onDrag(Offset point, Paint paint) {
+    if (isEditing) {
+      boundingBox.update(point, isRotating: isRotating);
+    }
+  }
+
+  @override
+  void onUp(Offset point, Paint paint) {
+    boundingBox.resetActiveCorner();
+  }
+
+  @override
+  void onCancel() {
+    currentPosition = null;
+    currentText = null;
+    isEditing = false;
+  }
+
+  @override
+  void onCheckmark(Paint paint) {
+    if (currentText != null) {
+      final textOffset = boundingBox.center;
+
+      final rectangleHeight =
+          (boundingBox.bottomLeft.dy - boundingBox.topLeft.dy).abs();
+
+      final textStyle = TextStyle(
+        color: paint.color,
+        fontSize: rectangleHeight / 2,
+      );
+
+      final command = commandFactory.createAddTextCommand(
+        textOffset,
+        currentText!,
+        textStyle,
+        paint,
+      );
+      commandManager.addGraphicCommand(command);
+      commandManager.clearRedoStack();
+      currentText = null;
+      currentPosition = null;
+      isEditing = false;
+    }
+  }
+
+  @override
+  void onPlus() {}
+
+  @override
+  void onRedo() {}
+
+  @override
+  void onUndo() {}
+
+  void drawGuides(Canvas canvas) {
+    drawRectangle(canvas, GraphicFactory.boundingBoxPaint, isBoundingBox: true);
+    _drawCornerCircles(canvas, GraphicFactory.anchorPointPaint);
+
+    final rectangleHeight =
```

Comment:
Painting text logic can be extracted to a public method and used in command painter

Reply:
Guides should be drawn after text.